templates/docs/examples/patterns/forms/_tristate_script.js

```diff
@@ -0,0 +1,31 @@
+/**
+  Sets up tri-state checkboxes and attaches event handlers to it
+  @param {String} selector The CSS selector matching tri-state checkboxes
+*/
+function initTristateCheckboxes(selector) {
+  var tristateCheckboxes = [].slice.call(document.querySelectorAll(selector));
+
+  tristateCheckboxes.forEach(function (checkbox) {
+    checkbox.addEventListener('change', function () {
+      var currentState = checkbox.getAttribute('aria-checked');
+
+      if (currentState === 'false') {
+        checkbox.setAttribute('aria-checked', 'mixed');
+        checkbox.checked = true;
```

Comment:
Technically it probably doesn't matter, but in MDN example they set `checked` to false when setting indeterminate to true: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox